src/tt.cpp

```diff
@@ -47,7 +47,7 @@ void TTEntry::save(Key k, Value v, Bound b, Depth d, Move m, Value ev) {
       key16     = (uint16_t)(k >> 48);
       value16   = (int16_t)v;
       eval16    = (int16_t)ev;
-      genBound8 = (uint8_t)(TT.generation8 | b);
+      genBound8 = (uint8_t)(TT.generation8 | PvNode << 2 | b);
```

Comment:
Can we really shift a bool like this without casting it first?

Reply:
As far as I can tell, the bool get promoted to an int before the shift. Also, my compiler (gcc) doesn't seem to mind. I wouldn't be against a more explicit formulation though.